src/components/authenticationPage.js

```diff
@@ -1,40 +1,32 @@
-import React, {Component} from 'react';
+import React, {Component, Redirect} from 'react';
+import {connect} from '../api.js'
 
 class WaitingPage extends Component {
     
-async postData(url = '', data = {},token) {
-    const response = await fetch(url, {
-      method: 'POST', 
-      headers: {
-        'Content-Type': 'application/json',
-        'Authorisation':token
-      },
-      redirect: 'follow',
-      referrerPolicy: 'no-referrer', 
-      body: JSON.stringify(data) 
-      });
-    return response.json(); 
-    } 
     getUrlVars = (vars) =>   {
-        window.location.href.replace(/[?&]+([^=&]+)=([^&]*)/gi, function(m,key,value) {
+        window.location.href.replace(/[?&#]+([^=&]+)=([^&]*)/gi, function(m,key,value) {
```

Comment:
Same as above, also as it's used in multiple places it should be extracted to a separate file and imported

Reply:
Again the qs will make this redundant